common/network-yarn/pom.xml

```diff
@@ -110,6 +110,7 @@
                 <exclude>META-INF/*.SF</exclude>
                 <exclude>META-INF/*.DSA</exclude>
                 <exclude>META-INF/*.RSA</exclude>
+                <exclude>META-INF/LICENSE</exclude>
```

Comment:
```
[WARNING] jackson-annotations-2.15.2.jar, jackson-core-2.15.2.jar, jackson-databind-2.15.2.jar, jackson-module-scala_2.13-2.15.2.jar, log4j-1.2-api-2.20.0.jar, log4j-api-2.20.0.jar, log4j-core-2.20.0.jar, log4j-slf4j2-impl-2.20.0.jar, spark-common-utils_2.13-4.0.0-SNAPSHOT.jar, spark-network-common_2.13-4.0.0-SNAPSHOT.jar, spark-network-shuffle_2.13-4.0.0-SNAPSHOT.jar, spark-network-yarn_2.13-4.0.0-SNAPSHOT.jar define 1 overlapping resource: 
[WARNING]   - META-INF/LICENSE
```

From the compile log, `META-INF/LICENSE` might come from multiple jars.

Since this `exclude` is indiscriminate, in this case, all `META-INF/LICENSE` would be excluded. If it includes a non-Apache LICENSE, it would also be excluded. Would this be a problem @srowen ?

Of course, even without the exclude, there would still be issues currently because multiple files with the same name would overwrite each other, and it seems difficult to determine which `META-INF/LICENSE` would be retained in the end.

Reply:
It looks like we include the license if all of those already in the LICENSE file (or no action is required) so it seems OK here. While it's nice to retain the license buried in the artifact, we should be addressing all licenses in the top level LICENSE and license/ dir, rather than rely on this anyway

Reply:
Understood, thank you for your explanation @srowen